File: .semgrep/logging-pii.java
class LoggingPiiExamples {
    void unsafe(CaseData caseData, GeneralApplication application, Party respondent1,
                TelemetryClient telemetryClient) {
        // ruleid: civil.java.pii-accessor-to-log
        log.info("Applicant email: {}", caseData.getEmailAddress());

        // ruleid: civil.java.pii-accessor-to-log
        log.warn("Claim amount=" + caseData.getClaimAmount());

        // ruleid: civil.java.sensitive-object-to-log
        log.info("Application: {}", application);

        // ruleid: civil.java.sensitive-object-to-log
        MDC.put("caseData", caseData);

        // ruleid: civil.java.sensitive-object-to-log
        telemetryClient.trackTrace(application);

        // ruleid: civil.java.sensitive-object-to-log
        log.info("Respondent: {}", respondent1);

        // ruleid: civil.java.pii-accessor-to-log
        MDC.put("party", caseData.getPartyName());

        // ruleid: civil.java.pii-accessor-to-log
        telemetryClient.trackEvent("payment", Map.of("amount", caseData.getPaymentAmount()), null);
    }

    void safe(CaseData caseData, String caseId, String userId) {
        // ok: civil.java.pii-accessor-to-log
        log.info("Processing case {}", caseData.getCcdCaseReference());

        // ok: civil.java.sensitive-object-to-log
        log.info("Processing case {} for user {}", caseId, userId);

        // ok: civil.java.pii-accessor-to-log
        MDC.put("caseId", caseId);
    }

    void safeAfterSensitiveMutation(CaseData caseData, UserDetails userDetails) {
        caseData.setRespondentSolicitor1EmailAddress(userDetails.getEmail());
        String organisationId = OrgPolicyUtils.getRespondent1SolicitorOrgId(caseData);

        // ok: civil.java.pii-accessor-to-log
        log.info("Organisation {} updated for case {}", organisationId,
            caseData.getLegacyCaseReference());
    }
}

File: .semgrep/logging-pii.ts
function unsafe(logger: Logger, claim: Claim, telemetryClient: TelemetryClient): void {
  // ruleid: civil.typescript.pii-field-to-log
  logger.info(`Email: ${claim.emailAddress}`);

  // ruleid: civil.typescript.pii-field-to-log
  logger.warn('Amount', claim.claimAmount);

  // ruleid: civil.typescript.sensitive-object-to-log
  logger.info('Claim', claim);

  const paymentInfo = claim.paymentInfo;
  // ruleid: civil.typescript.sensitive-object-to-log
  telemetryClient.trackEvent(paymentInfo);

  // ruleid: civil.typescript.pii-field-to-log
  telemetryClient.trackTrace({message: claim.partyName});

  // ruleid: civil.typescript.raw-console
  console.log(claim.id);
}

function safe(logger: Logger, claim: Claim, caseId: string, userId: string): void {
  // ok: civil.typescript.pii-field-to-log
  logger.info(`Processing case ${claim.id}`);

  // ok: civil.typescript.sensitive-object-to-log
  logger.info('Processing case and user', caseId, userId);

  const paymentStatus = getStatus(claim.paymentReference);
  // ok: civil.typescript.pii-field-to-log
  logger.info('Payment state', paymentStatus.status);

  // ok: civil.typescript.pii-field-to-log
  logger.debug('Interest option', claim.interest.interestClaimOptions);
}
